core/o11y/src/opentelemetry.rs

```diff
@@ -52,16 +52,25 @@ where
     };
     resource.push(KeyValue::new(SERVICE_NAME, service_name));
 
+    let overriding_vars = ["OTEL_BSP_MAX_CONCURRENT_EXPORTS", "OTEL_BSP_MAX_QUEUE_SIZE"];
+    let batch_config = if overriding_vars.iter().any(|v| std::env::var_os(v).is_some()) {
+        opentelemetry_sdk::trace::BatchConfigBuilder::default()
```

Comment:
does BatchConfigBuilder takes the values from thee env variables internally? also do you assume that if one is set, the other is also set (since you use any() above) or so?

Reply:
Yes, as documented in its documentation: https://docs.rs/opentelemetry_sdk/latest/opentelemetry_sdk/trace/struct.BatchConfigBuilder.html#method.default.

> also do you assume that if one is set, the other is also set (since you use any() above) or so?

I genuinely just implemented what was the easiest to implement here. Honestly, I don't think it matters what specific behaviour is implemented here, if a person is overriding any of these environment variables, it is perfectly reasonable in my eyes to require them to override additional ones, especially given that discovery of these requires reading code and/or documentation anyway.